kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -375,17 +388,31 @@ public WriteModel<WindowDoc> insert(
     final var partitionSegments = kafkaPartitionToSegments.get(kafkaPartition);
 
     final long epoch = partitionSegments.epoch;
-    return new UpdateOneModel<>(
-        Filters.and(
-            Filters.eq(WindowDoc.ID, compositeKey(windowedKey)),
-            Filters.lte(WindowDoc.EPOCH, epoch)
-        ),
-        Updates.combine(
-            Updates.set(WindowDoc.VALUE, value),
-            Updates.set(WindowDoc.EPOCH, epoch)
-        ),
-        UPSERT_OPTIONS
-    );
+    if (retainDuplicates) {
+      return new UpdateOneModel<>(
```

Comment:
I think you want to do an insert, not an update here. This will update the existing timestamp+key if it exists, when really we want to only ever insert.